File: app/Models/Order.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Order extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'order_number',
        'total',
        'subtotal',
        'tax',
        'shipping_fee',
        'discount',
        'coupon_code',
        'payment_method',
        'payment_gateway',
        'payment_status',
        'shipping_address_id',
        'billing_address_id',
        'tracking_number',
        'courier',
        'status',
        'expected_delivery_date',
        'notes'
    ];

    protected $casts = [
        'total' => 'decimal:2',
        'subtotal' => 'decimal:2',
        'tax' => 'decimal:2',
        'shipping_fee' => 'decimal:2',
        'discount' => 'decimal:2',
        'expected_delivery_date' => 'date'
    ];

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function items()
    {
        return $this->hasMany(OrderItem::class);
    }

    public function shippingAddress()
    {
        return $this->belongsTo(Address::class, 'shipping_address_id');
    }

    public function billingAddress()
    {
        return $this->belongsTo(Address::class, 'billing_address_id');
    }

    public function trackingEvents()
    {
        return $this->hasMany(OrderTrackingEvent::class);
    }

    public function return()
    {
        return $this->hasOne(ReturnRequest::class);
    }

    public function transaction()
    {
        return $this->hasOne(Transaction::class);
    }

    // Scopes
    public function scopePending($query)
    {
        return $query->where('status', 'pending');
    }

    public function scopeProcessing($query)
    {
        return $query->where('status', 'processing');
    }

    public function scopeCompleted($query)
    {
        return $query->where('status', 'completed');
    }

    public function scopeCancelled($query)
    {
        return $query->where('status', 'cancelled');
    }

    // Helper methods
    public function getStatusLabelAttribute()
    {
        return [
            'pending' => 'Pending',
            'processing' => 'Processing',
            'packed' => 'Packed',
            'shipped' => 'Shipped',
            'delivered' => 'Delivered',
            'cancelled' => 'Cancelled',
            'returned' => 'Returned'
        ][$this->status] ?? $this->status;
    }

    public function canBeCancelled(): bool
    {
        return in_array($this->status, ['pending', 'processing']);
    }

    public function canBeReturned(): bool
    {
        return $this->status === 'delivered' && 
               !$this->return && 
               $this->created_at->diffInDays(now()) <= 7;
    }

    public function generateTrackingNumber()
    {
        $this->tracking_number = 'DM-' . strtoupper(uniqid());
        $this->save();
        return $this->tracking_number;
    }
}

File: app/src/main/java/pk/dailymart/customer/models/Order.kt
package pk.dailymart.customer.models

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class Order(
    val id: Int,
    @SerializedName("order_number")
    val orderNumber: String,
    @SerializedName("tracking_number")
    val trackingNumber: String?,
    val subtotal: Double,
    @SerializedName("shipping_fee")
    val shippingFee: Double,
    val tax: Double,
    val discount: Double,
    val total: Double,
    @SerializedName("payment_method")
    val paymentMethod: String,
    @SerializedName("payment_status")
    val paymentStatus: String,
    val status: String,
    @SerializedName("created_at")
    val createdAt: String,
    @SerializedName("expected_delivery_date")
    val expectedDeliveryDate: String?,
    @SerializedName("courier")
    val courier: String
)

data class OrderDetail(
    val id: Int,
    @SerializedName("order_number")
    val orderNumber: String,
    @SerializedName("tracking_number")
    val trackingNumber: String?,
    val subtotal: Double,
    @SerializedName("shipping_fee")
    val shippingFee: Double,
    val tax: Double,
    val discount: Double,
    val total: Double,
    @SerializedName("payment_method")
    val paymentMethod: String,
    @SerializedName("payment_status")
    val paymentStatus: String,
    val status: String,
    @SerializedName("created_at")
    val createdAt: String,
    @SerializedName("expected_delivery_date")
    val expectedDeliveryDate: String?,
    @SerializedName("courier")
    val courier: String,
    val items: List<OrderItem>,
    @SerializedName("tracking_events")
    val trackingEvents: List<TrackingEvent>,
    @SerializedName("shipping_address")
    val shippingAddress: Address,
    val return: ReturnResponse?
)

data class OrderItem(
    val id: Int,
    val product: Product,
    val quantity: Int,
    val price: Double,
    val total: Double
)

data class TrackingEvent(
    val status: String,
    val description: String,
    val location: String?,
    @SerializedName("event_time")
    val eventTime: String
)

data class TrackingResponse(
    @SerializedName("tracking_number")
    val trackingNumber: String,
    @SerializedName("current_status")
    val currentStatus: String,
    val courier: String,
    @SerializedName("expected_delivery")
    val expectedDelivery: String?,
    val events: List<TrackingEvent>
)

data class PlaceOrderRequest(
    @SerializedName("shipping_address_id")
    val shippingAddressId: Int,
    @SerializedName("payment_method")
    val paymentMethod: String,
    val gateway: String? = null,
    @SerializedName("coupon_code")
    val couponCode: String? = null,
    val notes: String? = null
)

data class ReturnRequest(
    val reason: String,
    val items: List<ReturnItem>,
    val images: List<String>? = null
)

data class ReturnItem(
    @SerializedName("order_item_id")
    val orderItemId: Int,
    val quantity: Int
)

data class ReturnResponse(
    val id: Int,
    val status: String,
    @SerializedName("return_amount")
    val returnAmount: Double,
    val reason: String,
    val items: List<ReturnItem>
)
